core/input/input.cpp

```diff
@@ -1712,15 +1709,28 @@ void Input::add_joy_mapping(const String &p_mapping, bool p_update_existing) {
 }
 
 void Input::remove_joy_mapping(const String &p_guid) {
+	int index_removed = 0;
+
 	for (int i = map_db.size() - 1; i >= 0; i--) {
 		if (p_guid == map_db[i].uid) {
 			map_db.remove_at(i);
+			index_removed = i;
+
+			if (i == fallback_mapping) {
+				fallback_mapping = -1;
+				WARN_PRINT_ONCE(vformat("Removed fallback joypad input mapping \"%s\". This could lead to joypads not working as intended.", p_guid));
+			}
 		}
 	}
+
 	for (KeyValue<int, Joypad> &E : joy_names) {
 		Joypad &joy = E.value;
+
 		if (joy.uid == p_guid) {
 			_set_joypad_mapping(joy, -1);
+		} else if (joy.mapping > index_removed) {
+			// The map_db update offset this joypad's mapping reference, update it:
+			_set_joypad_mapping(joy, map_db[joy.mapping - 1]);
 		}
```

Comment:
```suggestion
		} else if (joy.mapping == index_removed ){
			_set_joypad_mapping(joy, fallback_mapping);
		}
```

Reply:
This is equal to `if (joy.puid == _guid)`, so there should be no need for that else-if-statement